### DISCUSSION TITLE
facebook oauth failed , but google oauth is success

### DISCUSSION THREAD

**Discussion by bobwatcherx** [pocketbase/pocketbase#220] [Q&A]
My Code : https://github.com/bobwatcherx/testpocket.git
I'm having problems with facebook oauth login

I use https on my vite. facebook is not allowed to use http in meta developer redirect oauth , so I have to set vite to https .

after I set it to https , I managed to get the codeVerifier payload and others

but when call function
client.Users.authViaOAuth2
the response is error

**Comment by ganigeorgiev** [comment]
It seems that you are still passing a http redirect url:
- https://github.com/bobwatcherx/testpocket/blob/master/src/pages/Private.svelte#L7
- https://github.com/bobwatcherx/testpocket/blob/master/src/pages/Login.svelte#L10

Another thing maybe is that you are using [vite-plugin-basic-ssl](https://github.com/vitejs/vite-plugin-basic-ssl), which generates untrusted certificates so I'm not sure how that works in general (I'm not sure if Facebook do validations on the redirect url domains).
Alternative to that plugin is to use something like [ngrok](https://ngrok.com/)  which will generate a valid public https address to test your Facebook integration with localhost.

If none of the above works, you could start PocketBase with `--debug` flag and check the more detailed error log message from the console.
